portfolio/src/main/webapp/script.js

```diff
@@ -16,9 +16,21 @@
 function executeFunctions() {
     moveText();
     getMessages();
+    signIn();
     collapsible();
 }
 
+// Run serverlet to get url for log in
+function signIn() {
+    fetch('/log-in', {method: 'POST'}).then(html => html.text()).then(htmlText => {
+      document.getElementById("log-in").innerHTML = htmlText;
+      const htmlTextLen = htmlText.length;
+      const loggedIn = htmlText.substring(htmlTextLen-11,htmlTextLen-5);
```

Comment:
Maybe make this more robust by using a regex to compare the string to Logout

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/match

Reply:
I didn't want to use regex in case the user's email contained the key word however I would index that last few chars and use regex on those.

This may change however if I try using JSP

Reply:
changed implementation